lrlex/src/lib/lexer.rs

```diff
@@ -293,15 +293,16 @@ impl<
                             lexemes.push(Ok(Lexeme::new(tok_id, old_i, longest)));
                         }
                         None => {
-                            lexemes.push(Err(LexError::new(Span::new(old_i, old_i))));
+                            lexemes.push(Err(LexError::new(Span::new(old_i, old_i), None)));
                             break;
                         }
                     }
                 }
                 if let Some((target_state_id, op)) = &r.target_state {
                     let state = match self.get_start_state_by_id(*target_state_id) {
                         None => {
-                            lexemes.push(Err(LexError::new(Span::new(old_i, old_i))));
+                            // TODO: I can see an argument for lexing state to be either `None` or `Some(target_state_id)` here
+                            lexemes.push(Err(LexError::new(Span::new(old_i, old_i), None)));
```

Comment:
Hmm, is this case with the TODO normally reachable when using lrlex, e.g. for `state` to be None at this point wouldn't one of `LexErrorKind::UnknownStartState`, or `InvalidStartState` would have previously triggered, implying some internal inconsistency around target_state_id?

Quite possible i'm misreading this, but if so maybe some thing like  `assert_eq!(None, Some(target_state_id))`, because I'm not sure what the user could do with a `StartStateId` if it isn't possible to get a `StartState` from it here, which looks to be the case.

Reply:
This might be another way of asking if we should just `self.get_start_state_by_id(*target_state_id).unwrap()`?